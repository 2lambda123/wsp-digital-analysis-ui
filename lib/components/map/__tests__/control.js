/* global describe, it, expect, jest */

import renderer from 'react-test-renderer'
import React from 'react'

import Control from '../control'

describe('Component > Map > Control', () => {
  it('renders correctly', () => {
    const props = {
      addOpportunityLayerToMap: jest.fn(),
      addIsochroneLayerToMap: jest.fn(),
      center: {
        lat: 38.898,
        lon: -77.015
      },
      clearIsochroneResults: jest.fn(),
      fetchIsochrone: jest.fn(),
      geocoderApiKey: 'MAPZEN_SEARCH_KEY',
      isochroneCutoff: 3600,
      isFetchingIsochrone: false,
      isShowingIsochrone: false,
      removeIsochroneLayerFromMap: jest.fn(),
      removeOpportunityLayerFromMap: jest.fn(),
      setIsochroneCutoff: jest.fn(),
      setIsochroneLatLng: jest.fn()
    }

    const tree = renderer.create(
      <Control
        {...props}
        />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
