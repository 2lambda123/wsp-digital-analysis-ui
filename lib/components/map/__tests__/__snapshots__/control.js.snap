exports[`Component > Map > Control renders correctly 1`] = `
<Control
  addIsochroneLayerToMap={[Function]}
  center={
    Object {
      "lat": 38.898,
      "lon": -77.015,
    }
  }
  clearIsochroneResults={[Function]}
  fetchIsochrone={[Function]}
  geocoderApiKey="MAPZEN_SEARCH_KEY"
  isFetchingIsochrone={false}
  isShowingIsochrone={false}
  isochroneCutoff={3600}
  removeIsochroneLayerFromMap={[Function]}
  setIsochroneCutoff={[Function]}
  setIsochroneLatLng={[Function]}>
  <div
    className="MapControl">
    <div
      className="block">
      <a
        onClick={[Function]}
        title="Toggle isochrone">
        <Icon
          type="dot-circle-o">
          <i
            className="fa fa-dot-circle-o fa-fw " />
        </Icon>
      </a>
    </div>
    <a
      className="block"
      onClick={[Function]}
      title="Search for an address">
      <Icon
        type="search">
        <i
          className="fa fa-search fa-fw " />
      </Icon>
    </a>
  </div>
</Control>
`;
